tests/bson/typemap-008.phpt

```diff
@@ -0,0 +1,122 @@
+--TEST--
+MongoDB\Driver\Cursor::setTypeMap(): fieldPath typemaps without server
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; ?>
+--FILE--
+<?php
+
+class MyArrayObject extends ArrayObject implements MongoDB\BSON\Unserializable
+{
+    function bsonUnserialize(array $data)
+    {
+        parent::__construct($data, ArrayObject::ARRAY_AS_PROPS);
+    }
+}
+
+class MyWildCardArrayObject extends MyArrayObject {};
+
+$bson = \MongoDB\BSON\fromPHP( [
+    '_id' => 1,
+    'object' => [
+        'parent1' => [
+            'child1' => [ 1, 2, 3 ],
+            'child2' => [ 4, 5, 6 ],
+        ],
+        'parent2' => [
+            'child1' => [  7,  8,  9 ],
+            'child2' => [ 10, 11, 12 ],
+        ],
+    ],
+] );
+
+function fetch($bson, $typeMap = [])
+{
+    for ($i = 0; $i < 25000; $i++) {
```

Comment:
Testing for memory leaks, and so that we can use this as a benchmarking script. It's a good thing that I added this test (with 50000) in the first place, as it *did* find a memory leak bug.